File: service_examples/src/service_client.cpp
#include<ros/ros.h>
#include<stdlib.h>
#include<service_examples/custom_srv.h>
#include<iostream>

using namespace std;

int main(int argc,char **argv)
{
    ros::init(argc,argv,"service_client");
    ros::NodeHandle n;
    ros::Rate rate(10);
    ros::ServiceClient client = n.serviceClient<service_examples::custom_srv>("speed_check");
    service_examples::custom_srv ji;
    ji.request.srvB.linear.x = 5;
    client.waitForExistence();
    while (ros::ok())
    {
        if (client.call(ji) == true)
        {
            if (ji.response.srvC.data == true)
            {
            cout<<"move forward \n";
            }
        }

        
        ji.request.srvB.linear.x = rand()%111-100;
        rate.sleep();

    }


    return 0 ;
}

File: service_examples/src/service_client.py
#!/usr/bin/env python

import rospy
from service_examples.srv import custom_srv,custom_srvRequest,custom_srvResponse
import random




def main():
    rospy.wait_for_service('speed_check')
    client = rospy.ServiceProxy('speed_check',custom_srv)
    Srv = custom_srvRequest()
    rate = rospy.Rate(10)

    while(not (rospy.is_shutdown())):
        Srv.srvB.linear.x = random.randint(-100,10)
        try: 
            result = client.call(Srv)
            if result.srvC.data:
                print('move forward')
        except rospy.ServiceException as e:
            print(e)
        rate.sleep()


if __name__ == "__main__":
    rospy.init_node('service_clinet')
    try :
        main()
    except rospy.ROSException as e:
        print(e)
